artiq/coredevice/phaser.py

```diff
@@ -886,6 +958,30 @@ def trf_read(self, addr, cnt_mux_sel=0) -> TInt32:
         return self.trf_write(0x00000008 | (cnt_mux_sel << 27),
                               readback=True)
 
+    @kernel
+    def cal_vco(self):
+        """Start VCO calibration."""
+        self.trf_write(self.trf_mmap[1] | (1 << 31))
```

Comment:
Why do you write from the map here but from the readback elsewhere?

Reply:
I agree this should be readback. I encountered a bug I haven't got to the bottom of when using read-back for this method. I can investigate further if you believe this to be a major issue, it does prevent users from changing many VCO frequency settings

Reply:
Maybe there is something wrong with readback. But I don't know whether that's due to the chip or not. Unclear whether a register can actually be fully read back. https://github.com/quartiq/phaser/issues/17

Reply:
My initial impression was, that there was a compiler issue. Printing mmap and readback values gave the same result. However, the hardware responded differently when using one or the other (no printing)